src/thing/build.ts

```diff
@@ -117,6 +120,10 @@ export type ThingBuilder<T extends Thing> = {
   addDecimal: Adder<number, T>;
   addInteger: Adder<number, T>;
   addStringNoLocale: Adder<string, T>;
+  addStringEnglish: (
+    property: Parameters<typeof addStringWithLocale>[1],
+    value: Parameters<typeof addStringWithLocale>[2]
+  ) => ThingBuilder<T>;
```

Comment:
Couldn't this simply be of type `Adder<string, T>` ?

Reply:
Good question! I don't know (i.e., I don't understand these TypeScript types at all).
I was just cutting-and-pasting from the function directly below it (because my new function simply calls that existing one with one less parameter), i.e.,:
```
  addStringWithLocale: (
    property: Parameters<typeof addStringWithLocale>[1],
    value: Parameters<typeof addStringWithLocale>[2],
    locale: Parameters<typeof addStringWithLocale>[3]
  ) => ThingBuilder<T>;
```
So in effect, I don't know why half these functions have `: Adder<type, T>` and the other half have `:Parameters<typeof 'funcName'>[x]`. In other words, if you're right (i.e., we **_can_** just use `Adder<string, T>` here), then can't we use it on all these functions...?
```
  addTime: Adder<Time, T>;
  addDecimal: Adder<number, T>;
  addInteger: Adder<number, T>;
  addStringNoLocale: Adder<string, T>;
  addStringEnglish: (
    property: Parameters<typeof addStringWithLocale>[1],
    value: Parameters<typeof addStringWithLocale>[2]
  ) => ThingBuilder<T>;
  addStringWithLocale: (
    property: Parameters<typeof addStringWithLocale>[1],
    value: Parameters<typeof addStringWithLocale>[2],
    locale: Parameters<typeof addStringWithLocale>[3]
  ) => ThingBuilder<T>;
  addNamedNode: (
    property: Parameters<typeof addNamedNode>[1],
    value: Parameters<typeof addNamedNode>[2]
  ) => ThingBuilder<T>;
  addLiteral: (
    property: Parameters<typeof addLiteral>[1],
    value: Parameters<typeof addLiteral>[2]
  ) => ThingBuilder<T>;
```
But in fact, now I notice something: is the code I added wrong - i.e., should the function name be _reused_ in the `typeof [funcName]` part - i.e., should this code actually be like the other examples I just pasted above, and so my new function should be this instead:
```suggestion
  addStringEnglish: (
    property: Parameters<typeof addStringEnglish>[1],
    value: Parameters<typeof addStringEnglish>[2]
  ) => ThingBuilder<T>;
```
????

Reply:
`addStringWithLocale` and `addStringEnglish` having the same types for the first parameters ( a Thing, a property and a value), the current typing wasn't wrong, but indeed using `addStringEnglish` would be better. However, reusing the `Adder` type definition also works, and looks cleaner in my opinion. See ba5dd0cfdabec531f63efcd351b435d80210549c.